code/game/objects/items/stacks/stack.dm

```diff
@@ -231,7 +231,9 @@
 			to_chat(usr, "<span class='warning'>You haven't got enough [src] to build \the [R.title]!</span>")
 		return FALSE
 	var/turf/T = get_turf(usr)
-	if(R.window_checks && !valid_window_location(T, usr.dir))
+
+	var/obj/D = R.result_type
```

Comment:
This isn't an instance can you just type it as `var/D`

Reply:
Actually the `var/obj/D` is needed as else it doesn't compile due to the fact that the dir var is in the /obj/ type.

Reply:
@AnturK 

That doesn't sound right.

You can do `initial(D.dir)` but you can't typecast it as an instance because it's a type. You're just tricking the compiler into compiling.

Reply:
I have no idea what you're talking about this is exactly how you use initial on types.

Reply:
I have been told not to typecast things as instances (i think for confusion to others who are reading might think it's an actual item), but this works(tm) since it's doing `initial(D.blahblah)` in the later blocks.

Otherwise it would cause runtimes, it might be better to just put a pin on this and make it clear that's a fake object whenever this code gets touched again.

Reply:
unfortunately `initial(R.result_type.dir)` also doesn't work. The compiler doesn't know that the object has a dir variable. I hoped that I could do that at first but ¯\\\_(ツ)_/¯

Reply:
please be slow I have problems.

Thanks for explaining, sucks tho.